.ci/test-install-docs.sh

```diff
@@ -6,18 +6,239 @@
+	# where the script is run from within the just-deleted directory.
+	mkdir -p "$cwd" && cd "$cwd"
+}
+
+setup()
+{
+	source /etc/os-release || source /usr/lib/os-release
+
+	mkdir -p "${GOPATH}"
+
+	get_tests_repo
+
+	[ -e "$mgr" ] || die "cannot find $mgr"
+}
+
+# Perform a simple test to create a container
+create_kata_container()
+{
+	local -r test_name="$1"
+
+	local -r msg=$(info "Successfully tested ${test_name} on distro ${ID} ${VERSION}")
+
+	# Perform a basic test
+	sudo -E docker run --rm -i --runtime "kata-runtime" "${docker_image}" echo "$msg"
+}
+
+# Run the kata manager to "execute" the install guide to ensure the commands
+# it specified result in a working system.
+test_distro_install_guide()
+{
+	info "Installing system from the $ID install guide"
+
+	$mgr install-docker-system
+
+	$mgr configure-image
+	$mgr enable-debug
+
+	local mgr_name="${mgr##*/}"
+
+	local test_name="${mgr_name} to test install guide"
+
+	info "Install using ${test_name}"
+
+	create_kata_container "${test_name}"
+
+	# Clean up
+	$mgr remove-packages
+}
+
+# Apart from the distro-specific install guides, users can choose to install
+# using one of the following methods:
+#
+# - kata-manager ("Automatic" method).
+# - kata-doc-to-script ("Scripted" method).
+#
+# Testing these is awkward because we need to "execute" the documents
+# describing those install methods, but since those install methods should
+# themselves entirely document/handle an installation method, we need to
+# convert each install document to a script, then delete all the kata code
+# repositories. This ensures that when each install method script is run, it
+# does not rely on any local files (it should download anything it needs). But
+# since we're deleting the repos, we need to copy this script to a temporary
+# location, along with the install scripts this function generates, and then
+# re-exec this script with an option to ask it to run the scripts the previous
+# instance of this script just generated.
+test_alternative_install_methods()
+{
+	local -a files
+	files+=("installing-with-kata-manager.md")
+	files+=("installing-with-kata-doc-to-script.md")
+
+	local tmp_dir
+
+	# FIXME: delete after run, but only with "-t"!
+	tmp_dir=$(mktemp -d)
+
+	local script_file
+
+	local file
+
+	local tests_dir
+	tests_dir="${tmp_dir}/tests"
+
+	mkdir -p "${tests_dir}"
+
+	local -i num=0
+
+	# Convert the docs to scripts
+	for file in "${files[@]}"
+	do
+		num+=1
+
+		local file_path
+		local script_file
+		local script_file_path
+		local test_name
+
+		file_path="${script_dir}/../install/${file}"
+		script_file=${file/.md/.sh}
+
+		# Add a numeric prefix so the tests are run in the array order 
+		test_name=$(printf "%.2d-%s" "${num}" "${script_file}")
+
+		script_file_path="${tests_dir}/${test_name}"
+
+		info "Creating test script ${test_name} from ${file}"
+
+		bash "${doc_to_script}" "${file_path}" "${script_file_path}"
+	done
+
+	reexec_in_tmpdir "${tmp_dir}"
```

Comment:
This is confusing :P thanks for document it

Reply:
np.